From d1cc28f6b4e221d6f0c8d21af4c2f7de963bbe82 Mon Sep 17 00:00:00 2001
From: Zhou Peng-B04994 <eagle.zhou@nxp.com>
Date: Mon, 13 Nov 2017 12:00:38 +0800
Subject: [PATCH 2783/5242] MLK-16805 - [i.MX8MQ/Hantro]: Enable runtime bus
 frequency

commit  d8d413081bb6a9fc929e5ba350bbc1fa679a4f9e from
https://source.codeaurora.org/external/imx/linux-imx.git

Add operation to request/release bus in runtime suspend/resume functions

Signed-off-by: Zhou Peng-B04994 <eagle.zhou@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/hantro/hantrodec.c |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/mxc/hantro/hantrodec.c b/drivers/mxc/hantro/hantrodec.c
index 25d77fb..14f6fad 100755
--- a/drivers/mxc/hantro/hantrodec.c
+++ b/drivers/mxc/hantro/hantrodec.c
@@ -47,6 +47,7 @@
 #include <linux/platform_device.h>
 #include <linux/pm_runtime.h>
 #include <linux/clk.h>
+#include <linux/busfreq-imx.h>
 
 #ifdef CONFIG_DEVICE_THERMAL
 #include <linux/device_cooling.h>
@@ -1816,13 +1817,13 @@ static int hantro_resume(struct device *dev)
 }
 static int hantro_runtime_suspend(struct device *dev)
 {
-	//release_bus_freq(BUS_FREQ_HIGH);
+	release_bus_freq(BUS_FREQ_HIGH);
 	return 0;
 }
 
 static int hantro_runtime_resume(struct device *dev)
 {
-	//request_bus_freq(BUS_FREQ_HIGH);
+	request_bus_freq(BUS_FREQ_HIGH);
 	hantro_ctrlblk_reset(dev);
 	return 0;
 }
-- 
1.7.9.5

